src/finch/interface/lazy.py

```diff
@@ -179,6 +181,25 @@ def __bool__(self) -> bool:
         )
 
 
+def asarray(arr):
+    """
+    Return np.ndarray if arr is a scalar type value, otherwise return unchanged arr.
+
+    Parameters:
+    ------------
+    - arr: The input array to be converted into a LazyTensor.
+
+    Returns:
+    ------------
+    out: array
+        an output array converted from input parameter ``arr``.
+    """
+    if isinstance(arr, Number | str | bytes | bytearray | memoryview):
```

Comment:
Instead of an `if isinstance`, could we do a two step process where we first check for an `asarray` attribute, and if it doesn't exist we use `query_property(arr, "asarray", "__attr__")`?

Then we could do `register_property(Number, "asarray", "__attr__", lambda x: np.asarray(x))`.